### DISCUSSION TITLE
book/nodejs/

### DISCUSSION THREAD

**Discussion by giscus** [xsltdev/scriptdev.ru#125] [Announcements]
# book/nodejs/

Справочник Typescript с примерами кода

https://scriptdev.ru/book/nodejs/

**Comment by DenisDolzhikov** [comment]
build:live не работает, так же необходимо подшаманить buld, т.к всегда перед билдом желательно удалять папку lib/dist (где скомплированный js храните), чтобы не накапливались мусорные неиспользуемые файлы.
Вот мое решение:

// package.json
"scripts": {
    "build": "npm run rmdir dist && npx tsc",
    "start": "node dist/index.js",
    "dev": "nodemon src/index.ts",
    "rmdir": "node -e \"var fs = require('fs'); try{process.argv.slice(1).map((fpath) => fs.rmdirSync(fpath, { recursive: true 
         }))}catch(err){console.log(`Dist not found`)}; process.exit(0);\"",
}

// nodemon.json 
{
  "watch": ["src"],
  "ext": "ts",
  "exec": "concurrently \"npx tsc --watch\" \"ts-node src/index.ts\""
}
